projects/hello-cmake-package/CMakeLists.txt

```diff
@@ -0,0 +1,127 @@
+
+project(hello
+    VERSION "0.1.0"
+)
+
+include(GNUInstallDirs)
+
+# define the C++ library "hello"
+add_library(hello SHARED
+    "${PROJECT_SOURCE_DIR}/src/hello.cpp"
+)
+
+target_include_directories(hello
+PUBLIC
+    $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
+    $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
+)
+
+install(DIRECTORY "${PROJECT_SOURCE_DIR}/include/"
+    DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
+)
+
+# Standard installation subdirs for the C++ library are used. The files will end up in
+# the specified subdirectories under the Python package root.
+# For example, "<python package prefix>/hello/lib/" if the destination is "lib/".
+#
+# Installing the objects in the package provides encapsulation and will become important
+# later for binary redistribution reasons.
+install(TARGETS hello
+    EXPORT helloTargets
+    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+)
+
+# The CMake package config and target files are installed under the Python package root.
+# This is necessary to ensure that all the relative paths in the helloTargets.cmake
+# resolve correctly. It also provides encapsulation.
+#
+# The actual path used must be selected so that consuming projects can locate it via
+# `find_package`. To support finding CMake packages in the Python package prefix,
+# using `find_package`s default search path of `<prefix>/<name>/share/<name>*/cmake/`
+# is reasonable. Adding the Python package installation prefix to CMAKE_PREFIX_PATH
+# in combination with this path will allow `find_package` to find this package and any
+# other package installed via a Python package if the CMake and Python packages are
+# named the same.
+set(HELLO_CMAKE_PACKAGE_INSTALL_SUBDIR "share/hello/cmake")
+
+install(EXPORT helloTargets
+    NAMESPACE hello::
+    DESTINATION ${HELLO_CMAKE_PACKAGE_INSTALL_SUBDIR}
+)
+
+include(CMakePackageConfigHelpers)
+
+write_basic_package_version_file(
+    helloConfigVersion.cmake
+    VERSION ${PROJECT_VERSION}
+    COMPATIBILITY SameMinorVersion
+)
+
+configure_package_config_file(
+    "${PROJECT_SOURCE_DIR}/cmake/helloConfig.cmake.in"
+    helloConfig.cmake
+    INSTALL_DESTINATION ${HELLO_CMAKE_PACKAGE_INSTALL_SUBDIR}
+)
+
+install(FILES
+    "${PROJECT_BINARY_DIR}/helloConfig.cmake"
+    "${PROJECT_BINARY_DIR}/helloConfigVersion.cmake"
+    DESTINATION ${HELLO_CMAKE_PACKAGE_INSTALL_SUBDIR}
+)
+
+# We are using the SKBUILD variable, which is defined when scikit-build is running the
+# CMake build, to control building the Python wrapper. This allows the C++ project to be
+# installed, standalone, when using the standard CMake build flow.
+if (DEFINED SKBUILD)
+
+    # prevent an unused variable warning
+    set(ignoreMe "${SKBUILD}")
+
+    # call pybind11-config to obtain the root of the cmake package
+    execute_process(COMMAND ${PYTHON_EXECUTABLE} -m pybind11 --cmakedir OUTPUT_VARIABLE pybind11_ROOT_RAW)
+    string(STRIP ${pybind11_ROOT_RAW} pybind11_ROOT)
+    find_package(pybind11)
```

Comment:
I'm hoping to remove the need to do this eventually. :)

Reply:
How's that?

Reply:
Current idea is to add an entry point mechanism to scikit-build. Pybind11 would define the entrypoint:

```ini
[options.entry_points]
scikit-build.module =
  module=pybind11.skbuild:module
```

The "module" function returns a path pointing to the CMake files in the package. This is added to CMake’s search path by scikit-build, iterating over all matching entry points.

Reply:
Hmmm... I'm not familiar with the tooling around discovering entry points, but it sounds reasonable. Are those CMake files to be added via something akin to `add_subdirectory`? I have a project I'm writing in C++, but also want to distribute a Python wrapper (using pybind11) and distribute the C++ libraries, CMake to consume those libraries, and the Python wrapper using pip.

I did make the changes suggested. Are we waiting on @jcfr to do a final review and pull?